### ISSUE TITLE
Add Feedback Form running inside Docker container for Meetups, Training etc.

### ISSUE THREAD

**Issue by ajeetraina** [franela/play-with-docker#116]
Play with Docker is an ideal platform now for training sessions, Meetups & workshops. How about gathering feedback in a "Docker" Way. At the end of session, the organizer or mentor can ask audience or students to pull and run a docker container which opens up a web page listening at specific port and get a feedback. The feedback can be pushed to a centralized database or instantly be displayed under certain section of PWD website.

Its always good to know how was the experience, what worked and what didn't work for the particular meetup, conference and training events along with room for improvement.

**Comment by xetorthio** [comment]
Yes. I like the idea of having feedback. I wonder if we should do it with
the container you mentioned or with a Google form survey.

**Comment by marcosnils** [comment]
I also like the idea, not sure about the container approach. It could be "cool", but it might confuse the users. 

I don't have a lot of experience in designing UI's and getting user's feedback, so I'm not sure which would be the best way to proceed.

**Comment by mikesir87** [comment]
If we decided to go the non-container route and do it straight from the UI, I'd be happy to out something together. Question... Where's the feedback going?

**Comment by marcosnils** [comment]
It'd be great to have @vcoisne and @manomarks expertise on feedback gathering for this.

**Comment by ajeetraina** [comment]
I think of this as a simple approach:

1. $docker run -d -p 8001:8001  ajeetraina/pwdfeedback

It will display port number 8001 on the top of the page. User will click on it to open up a HTML page, answer the queries and submit. It gets push to the centralized location/DB for further analysis.

**Comment by marcosnils** [comment]
@ajeetraina not sure if that will work. I think I prefer to stick to a standard feedback component in the same page instead of doing something crazy. I mean.. running a container just to send feedback might be confusing and error prone.

**Comment by mikesir87** [comment]
If were to go the container approach, I recommend having the container run a command that interactively gets the feedback using console input.  Something like...

```
docker run -ti pwdfeedback
> How can we make it better?
[ User enters feedback here ]
```

To help make it less confusing on how to provide feedback, we could have a button in the bottom-right corner of the page that says "Provide feedback" that, when clicked, inserts the command to run into the console.

One advantage of using a container is it could make it easier to send the email, etc. without relying on a third-party service to handle a form request (as PWD is a static site).  Would think about how that container would be configured with where to send the note, etc. and how to get the image on the node.  Do we want it in a remote repo?  If so, is that a public repo?  If not, where does it go?  One option could be to push it to ECR and authorize the node to pull from ECR using IAM.

The advantages of staying in the browser are 1) it's familiar to most users, 2) you can present the entire form at once, allowing users know what questions they can answer, and 3) we could potentially get screenshot grabs (what section isn't working as expected/needs correction?).  We would then have to figure out how the feedback is actually sent (where does the form post _actually_ go?).

**Comment by ManoMarks** [comment]
So we've had a lot of success on docs.docker.com with a feedback button at the bottom of the page. You can see it here:
https://docs.docker.com/docker-for-mac/
As well as buttons for:
* Edit this page
* Request docs changes
Since we're planning on porting to a docs frame, this should come along for free.
